### DISCUSSION TITLE
How to sort a gallery, so that new images are automatically at the beginning/ at the top of the post

### DISCUSSION THREAD

**Discussion by michaelfluegge** [GetPublii/Publii#1758] [Help] [answered]
I sometimes add an image to a gallery, and these images are placed at the bottom of the gallery. Is there an option to choose/to activate that new images are placed as first image of the gallery automatically? I looked at the settings and the documentation of Publii and Photoswipe but didn't find anything suitable.

**Comment by bjazmoore** [comment]
Maybe ask this question on the photoswipe forum?

**Reply by michaelfluegge** [reply]
As I can sort the images manually using drag and drop after import, I thought there an option to sort automatically which I oversaw. Anyway, I will try the photoswipe forum :-) , thanks

**Reply by candidexmedia** [reply]
@michaelfluegge let us know if you find a solution. I'm also curious...

**Reply by michaelfluegge** [reply]
@candidexmedia, I had some conversation with chatGTP, and I understood that it would be possible to build a plugin with some javascript in it to move the latest image in the HTML to the top of the list (within the post editor). 
But building a plugin would be too much new stuff for me. I am more into using Publii (or tweaking it a bit) rather than inventing it new.

**Reply by michaelfluegge** [reply]
Ok, @bjazmoore , I had a deeper look into this, and it is not the PhotoSwipe part. When adding an image to a gallery, Publii adds the HTML automatically at the end. 
So I can cut the <figure></figure> part of the new image and paste it at the beginning of the HTML, or move it using the arrows to the top. Both is very unhandy, especially when there is larger size of images in the gallery.

Sort options for the gallery by name/date/date added/etc like in other tools would be cool.

**Comment by gpsblues** [comment, marked as answer]
Hi! To place new images at the top of your gallery, try using this CSS:

```
.gallery {
    flex-wrap: wrap-reverse;     /* Places new images at the top */
    flex-direction: row-reverse;  /* Reverses the order of images */
}

```

This way, new images will automatically appear first. Give it a try!"

**Reply by candidexmedia** [reply]
@gpsblues I was just coming in to suggest the same! I tested your sample on the Simple Theme demo, and it works perfectly! 😊

**Reply by michaelfluegge** [reply]
It is not I was looking for, but at least the result is what I was expecting :-) 
Thanks, @gpsblues and @candidexmedia

**Comment by michaelfluegge** [comment]
@gpsblues , @candidexmedia , I just realized that your CSS goes conflict in a way with Bob's suggestion from #1917 : 
.gallery__item {
   flex-grow: 0 !important;
}
When using your CSS, I have one image in the last row (which is at the top now) only :-( 

<img width="1390" alt="Bildschirmfoto 2025-01-28 um 22 14 33" src="https://github.com/user-attachments/assets/be7d7f27-b1bc-4485-9304-4bb6eb6c7fa4" />

**Comment by gpsblues** [comment]
Hi @michaelfluegge,

The logic behind the solutions we proposed is to modify the CSS to change the display order of the elements by inverting them. On the Simple theme, everything seems to work fine, but other themes might use a different organization of flex elements or even a grid layout (I’m not sure about this), so the rules may need to be refined or adjusted accordingly.

Normally, the first image (which is the last one in the standard flow) should take up 100% of the width.

You might have some additional rules that are restricting the element’s size or enforcing fixed-size thumbnails. Alternatively, there could be extra rules forcing specific element dimensions.

To ensure the image properly expands to fill the available space, you could try adding flex-grow: 1; to the relevant element.

**Comment by candidexmedia** [comment]
@michaelfluegge I know you marked this as resolved, but I still think having a sort function in the Gallery GUI of the post editors would be a worthy additon for people who don't know CSS.

**Reply by michaelfluegge** [reply]
@candidexmedia , I fully agree, but Bob and ‚Dzuidek‘ are ignoring all request regarding photographers needs, like EXIF import to Caption, Gallery sort, or an interface to LrC (which I extremely miss from my Wordpress experience)

**Reply by candidexmedia** [reply]
Hmmm, well hopefully, it can catch the eye of someone in the community with some web dev skills, especially since Publii offers a photography theme. I've seen the community work on some pretty cool workarounds in the last years (b0fh/Chris's guidance on the TinyMCE Template Plugin, for example, has been a godsend).

What's LrC?

**Reply by michaelfluegge** [reply]
Oh, sorry, LrC is Lightroom Classic :-) . And there are more than one photography theme, or? One free and three paid.
